mimic/rest/maas_api.py

```diff
@@ -518,9 +518,21 @@ def delete_entity(self, request, tenant_id, entity_id):
         Delete an entity, all checks that belong to entity, all alarms that belong to those checks
         """
         entities = self._entity_cache_for_tenant(tenant_id).entities_list
+        previous_size = len(entities)
         entities[:] = [entity for entity in entities
                        if entity.id != entity_id]
 
+        if len(entities) == previous_size:
```

Comment:
This seems non intuitive. Why compare the lengths of the list instead of trying to find the item in `self._entity_cache_for_tenant(tenant_id).entities_list`.